### PR TITLE
[ENH] - Update cycle approach

### PR DESCRIPTION
**Pull request by TomDonoghue**
This started as a quick update to cycles, and snowballed a little into some slightly bigger updates / orgs, but I think it all makes sense. 

Changes:
- update the approach for `sim_cycle`, so that it grabs available cycle functions, rather an explicit conditionals
    - to make this approach consistent, each cycle was updated to be it's own function
- separating out cycles from transients
    - this came along as part of the above, and I think is a sensible / useful conceptual organization
- parameter checking of inputs for cycles (related to #195)
    - this lead to some overall updates to checker code
- update to be able to phase shift cycles, to be able to have variable start phases for oscillation sims (#187)

Review notes:
- @ryanhammonds - if you can review for code, as normal.
- @elybrand - I tagged you for checking if this matches what you were thinking in #195 for checking sim params
    - also, could you sanity check the phase shift approach. No mathematical suprises on this?
- @rdgao - if you can sanity check, does the slightly updated organization make sense to you?

### CODE REVIEW

File: neurodsp/sim/cycles.py [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+from scipy.signal import gaussian, sawtooth
+
+from neurodsp.utils.checks import check_param
+from neurodsp.sim.info import get_sim_func
+from neurodsp.sim.transients import sim_synaptic_kernel
+
+###################################################################################################
+###################################################################################################
+
+def sim_cycle(n_seconds, fs, cycle_type, phase=0, **cycle_params):
```
Comment (ryanhammonds) [inline neurodsp/sim/cycles.py:13]: `phase` isn't used in this func.

File: neurodsp/sim/info.py [hunk 2]
```diff
@@ -23,6 +21,8 @@ def get_sim_funcs(module_name):
         A dictionary containing the available sim functions from the requested sub-module.
     """
 
+    from neurodsp.sim import periodic, aperiodic, transients, combined, cycles
```
Comment (ryanhammonds) [inline neurodsp/sim/info.py:24]: Why were these imports moved to inside the func?

File: neurodsp/sim/cycles.py [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    Parameters
+    ----------
+    n_seconds : float
+        Length of cycle window in seconds.
+        This is NOT the period of the cycle, but the length of the returned array of the cycle.
+    fs : float
+        Sampling frequency of the cycle simulation.
+    cycle_type : {'sine', 'asine', 'sawtooth', 'gaussian', 'exp', '2exp'}
+        What type of cycle to simulate. Options:
+
```
Comment (elybrand) [inline neurodsp/sim/cycles.py:25]: There is no docstring for `phase`.

File: neurodsp/sim/cycles.py (hunk 1 above)
Comment (TomDonoghue) [reply neurodsp/sim/cycles.py:13]: ahh, good catch! At one point I applied phase rotation here, but I think it makes more sense in sim_periodic. I've dropped this argument from here.

File: neurodsp/sim/info.py (hunk 2 above)
Comment (TomDonoghue) [reply neurodsp/sim/info.py:24]: Circular imports! All hell breaks loose if they are at the top (it breaks). 

This function is programmatically quirky (not ideal, in best practices sense), but it's efficient at what it does, so I sorta like it.

File: neurodsp/sim/cycles.py [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+    Returns
+    -------
+    cycle : 1d array
+        Simulated sine cycle.
+
+    Examples
+    --------
+    Simulate a cycle of a 1 Hz sine wave:
+
+    >>> cycle = sim_asine_cycle(n_seconds=1, fs=500)
```
[the remark this answers is not in the document]
Comment (TomDonoghue) [reply neurodsp/sim/cycles.py:89]: Ooops, yeh, copy paste errors on the docstrings. Fixed.

File: neurodsp/sim/cycles.py (hunk 3 above)
Comment (TomDonoghue) [reply neurodsp/sim/cycles.py:25]: Ah, phase in this func was accidentally leftover from when I was playing with where to apply phase shifts. I've dropped it from this func, and removed it from the signature.

**Pull request merged**